meson.build

```
@@ -34,7 +34,8 @@ openssl_dep = dependency('openssl', version: '>=1.1.1', required: get_option('cp
 if openssl_dep.found()
   deps += openssl_dep
   args += '-DCPPHTTPLIB_OPENSSL_SUPPORT'
-  if host_machine.system() == 'darwin'
+  if get_option('cpp-httplib_apple_keychain') and host_machine.system() == 'darwin'
+    args += '-DCPPHTTPLIB_USE_CERTS_FROM_APPLE_KEYCHAIN'
     deps += dependency('appleframeworks', modules: ['CoreFoundation', 'Security'])
   endif
```

> So it is strange when linker gets additional flags without any semantic artifacts in the build script

I'm not sure what you mean by this, but if I understand correctly then no, it is not strange to me. It is perfectly reasonable for header-only libraries to depend on third party libraries, and in that case the build system should add the required compiler and linker flags to the interface of the header-only library, so that consuming the header from another project doesn't result in linker errors. Did I misunderstand your comment?

Anyway, thinking a bit more about this I think that it'd be better to fully make use of Meson's `feature` type just like what we already do with OpenSSL, Zlib, etc. I'd suggest something like this:

```suggestion
  if host_machine.system() == 'darwin'
    apple_keychain_dep = dependency('appleframeworks', modules: ['CoreFoundation', 'Security'], required: get_option('cpp-httplib_apple_keychain'))
    if apple_keychain_dep.found()
        deps += apple_keychain_dep
        args += '-DCPPHTTPLIB_USE_CERTS_FROM_APPLE_KEYCHAIN'
    endif  
  endif
```

---

> I would keep the Meson (and CMake) code as is, as these libraries are always available on macOS (right?)

I just meant that it seems reasonable to add optionality for the feature both in the header using a macro and in build systems. Nothing else.